ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/util/RowUtil.java

```diff
@@ -50,4 +51,69 @@ private static ColumnType columnTypeFromString(final String columnType) {
     ));
     return new ColumnTypeImpl(primaryType);
   }
+
+  public static List<String> colNamesFromSchema(final String schema) {
+    return splitAcrossOneLevelDeepComma(schema).stream()
+            .map(RowUtil::removeBackTickAndKeyTrim)
+            .map(RowUtil::splitAndGetFirst)
+            .collect(Collectors.toList());
+  }
+
+  public static List<String> colTypesFromSchema(final String schema) {
+    return splitAcrossOneLevelDeepComma(schema).stream()
+            .map(RowUtil::removeBackTickAndKeyTrim)
+            .map(RowUtil::splitAndGetSecond)
+            .collect(Collectors.toList());
+  }
+
+  private static List<String> splitAcrossOneLevelDeepComma(final String stringWithCommas) {
+    final List<String> listCols = new LinkedList<>();
+    final StringBuilder sb = new StringBuilder();
+    int nestRound = 0;
+    int nestChevron = 0;
+    for (int i = 0; i < stringWithCommas.length(); ++i) {
+      final char ch = stringWithCommas.charAt(i);
+      if (i == stringWithCommas.length() - 1) {
+        sb.append(ch);
+        listCols.add(sb.toString());
+        break;
+      }
+      switch (ch) {
+        case ',':
+          if (nestRound == 0 && nestChevron == 0) {
+            listCols.add(sb.toString());
+            sb.setLength(0);
+            continue;
+          }
+          break;
+        case '(':
+          ++nestRound;
+          break;
+        case ')':
+          --nestRound;
+          break;
+        case '<':
+          ++nestChevron;
+          break;
+        case '>':
+          --nestChevron;
+          break;
+        default:
+      }
+      sb.append(ch);
+    }
+    return listCols;
+  }
+
+  private static String removeBackTickAndKeyTrim(final String dirtyString) {
+    return dirtyString.replace(" KEY", "").replace("`", "").trim();
+  }
+
+  private static String splitAndGetFirst(final String stringToSplit) {
+    return stringToSplit.split(" ", 2)[0];
```

Comment:
Maybe this is always going to work, but you could make this more of a generic whitespace matcher rather than a single space.  Anyway, not a huge deal so long as we have unit tests that are going to catch any changes.

Reply:
let's keep it specific for now as it is highly dependent on our schema format today (and it is made for one and only one purpose)